proto/proto/dex.proto

```diff
@@ -12,28 +12,66 @@ message Swap {
   MockFlowCiphertext enc_amount_1 = 2;
   // Encrypted amount of asset 2 of the trading pair.
   MockFlowCiphertext enc_amount_2 = 3;
+  // Encapsulates the authorized fields of the Swap transaction, used in signing.
+  SwapBody body = 4;
+  // The spend authorization signature is effecting data.
+  crypto.SpendAuthSignature auth_sig = 5;
```

Comment:
I don't think we need a spend auth signature here, because the `Swap` action consumes the input funds from the transaction's value balance, rather than spending any notes itself.  So the `Swap` will have to be alongside a `Spend` action(s) that unlock the required input funds, and those actions will have the spend authorization signatures for each of those notes already.